opencensus/proto/stats/metrics/metrics.proto

```diff
@@ -215,12 +215,30 @@ message DistributionValue {
     // The number of values in each bucket of the histogram, as described in
     // bucket_bounds.
     int64 count = 1;
-
-    // TODO: Add support for exemplars.
   }
 
   // If the distribution does not have a histogram, then omit this field.
   // If there is a histogram, then the sum of the values in the Bucket counts
   // must equal the value in the count field of the distribution.
   repeated Bucket buckets = 6;
+
+  // Exemplars are example points that may be used to annotate aggregated
+  // Distribution values. They are metadata that gives information about a
+  // particular value added to a Distribution bucket.
+  message Exemplar {
+    // Value of the exemplar point.
+    int64 value = 1;
+
+    // The observation (sampling) time of the above value.
+    google.protobuf.Timestamp timestamp = 2;
+
+    // The trace id associated with the example value.
+    bytes trace_id = 3;
```

Comment:
What is the reason for supporting trace_id/span_id instead of arbitrary attachments like:  https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries#Exemplar

I think there's lots of value in having other data associated with exemplars, for example the full tag set or maybe a stack trace.

Reply:
In my opinion the advantage of having strong type instead of any is 1) we don't need to worry about encoding/decoding here, and 2) we can be sure that only a single attachment of any given message type is in a single exemplar, which is enforced by Stackdriver backend. If we want to have other attachments, we can add additional fields to Exemplar.

BTW Stackdriver doesn't really support arbitrary attachments in Exemplar. For now if we try to attach some message other than span context or dropped label to Exemplar, Stackdriver backend will return an error.

Reply:
The advantage of Any is that the backend can just store the items without actually interpreting them and return them in the API that the application can interpret any way it likes. The fact that Stackdriver enforces a particular kind seems to cut off useful applications unnecessarily. 

Regarding the validation of only a single type per Exemplar, I thought that we had decided that we would not try to do validation in the metrics proto? 

As an alternative to Any, What about changing this to be a `oneof` of SpanContext and dropped tags?

Reply:
Using `oneof` sounds better, I updated the attachment proto. 

I agree that we should avoid doing validations, though I think it makes sense to have restriction on the type of attachments we want to Exemplar.